src/app/[locale]/filter-set.tsx

```diff
@@ -0,0 +1,65 @@
+import {SearchResultFilter} from '@/lib/dataset-fetcher';
+import {useCallback, Dispatch} from 'react';
+
+interface Props {
+  title: string;
+  searchResultFilters: SearchResultFilter[];
+  selectedFilters: string[];
+  setSelectedFilters: Dispatch<string[]>;
+}
+
+export default function FilterSet({
+  title,
+  searchResultFilters,
+  selectedFilters,
+  setSelectedFilters,
+}: Props) {
+  const handleChange = useCallback(
```

Comment:
For consideration (not something we need to solve now): perhaps we can use a headless component/primitive for rendering and handling the checkboxes (e.g. Radix UI)? This could save us some boilerplate code.

Reply:
Yes, This is possible. I will take a look at this on the next tickets.